### ISSUE TITLE
WeatherKit

### ISSUE THREAD

**Issue by 1ec5** [ceeK/Solar#60]
WeatherKit has a [`CurrentWeather.isDaylight`](https://developer.apple.com/documentation/weatherkit/currentweather/isdaylight) property and [`SunEvents`](https://developer.apple.com/documentation/weatherkit/sunevents) struct that essentially replicate the functionality of this library. However, WeatherKit is only available for this year’s operating system releases (iOS 16, iPadOS 16, macOS 13, tvOS 16, watchOS 9). Should this library somehow integrate with WeatherKit on supported platforms, perhaps becoming a compatibility shim for it on unsupported Swift platforms like Linux?

**Comment by 1ec5** [comment]
This might be a tad optimistic, considering potential pricing and usage limits around WeatherKit in general, but I opened this issue in case the presumably local stuff above is exempt.

**Comment by gshaw** [comment]
WeatherKit also requires a network connection and as isn't free.  Personally, I think there is value in having a focused Swift package like Solar for those that need sunrise and sunset but not the weather.

**Comment by 1ec5** [comment]
Yeah, it’s unfortunate that the calendrical functionality got bundled up with the live weather functionality. Maybe a good next step would be to investigate and document how well this library and WeatherKit agree on the sunrise/sunset calculations?
